### PR TITLE
added enzyme tests for Login component

### PR DESCRIPTION
**Pull request by dinodsaurus**
<br/><br/><br/><url>LiveURL: https://kiwicomsmart-faq-dmfxbhqwkp.now.sh
</url>
</url>
</url>

### CODE REVIEW

File: src/KiwiLogin/index.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ type State = {|
   showError: boolean,
 |};
 
-class KiwiLogin extends React.Component<Props, State> {
+export class PureKiwiLogin extends React.Component<Props, State> {
```
Comment (dinodsaurus) [inline src/KiwiLogin/index.js]: We can discuss about naming of the pure components which are exported without HOC stuff

Comment (dinodsaurus) [reply src/KiwiLogin/index.js]: I saw you call it `RawComponent,` so I will follow that naming convention ;)

File: src/KiwiLogin/__test__/KiwiLogin.test.js [hunk 2]
```diff
@@ -1,23 +1,68 @@
...
+    });
+
+    expect(Login.state).toMatchSnapshot();
+  });
+
+  it('should handle sign in', async () => {
+    const Login = wrapper.instance();
+    await Login.handleSignIn(e);
+    expect(e.preventDefault.mock.calls).toHaveLength(1);
+    expect(Login.props.history.push.mock.calls[0][0]).toBe('/content');
```
Comment (git-toni) [inline src/KiwiLogin/__test__/KiwiLogin.test.js]: `/content` ----> `routeDefinitions.CONTENT` from `/src/routeDefinitions`.

File: src/KiwiLogin/__test__/KiwiLogin.test.js [hunk 3]
```diff
@@ -1,23 +1,68 @@
...
-      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
-        <KiwiLogin store={store} />
-      </MemoryRouter>
-    </CloseContext.Provider>
-  );
-  */
+  const props = {
+    onLogin: () =>
+      new Promise(resolve => {
+        process.nextTick(() => resolve());
```
Comment (git-toni) [inline src/KiwiLogin/__test__/KiwiLogin.test.js]: Just curious, why not just `new Promise(resolve => resolve())`, wouldn't it end up basically at the same queue position? :slightly_smiling_face:  Idem for Line 51 below.

File: src/KiwiLogin/__test__/KiwiLogin.test.js (hunk 2 above)
Comment (dinodsaurus) [reply src/KiwiLogin/__test__/KiwiLogin.test.js]: I personally like to have stuff hardcoded in tests, so they fail if I make some changes. But if you think this is a better approach I don't mind changing it 👍

Comment (git-toni) [reply src/KiwiLogin/__test__/KiwiLogin.test.js]: But aren't you testing that the history was pushed **exactly** to `routeDefinitions.CONTENT`, as expected behavior from the component?

File: src/KiwiLogin/__test__/KiwiLogin.test.js (hunk 3 above)
Comment (dinodsaurus) [reply src/KiwiLogin/__test__/KiwiLogin.test.js]: There is no specific reason behind this, I just always do it like this..
and now when you asked I realized it doesnt make sense 😂
I think the cleanest approach would be `async () => {}`

Comment (git-toni) [reply src/KiwiLogin/__test__/KiwiLogin.test.js]: Haha, fair enough. In your defense, jest's docs also use `nextTick` thing. 

`async () => {}` is nice but isn't `onLogin` specifically a `Promise` as received by props? https://github.com/kiwicom/smart-faq/blob/6e8fe3ad41bf65ab01ab47611e51ac99cb69cdec/src/types.js

File: src/KiwiLogin/__test__/KiwiLogin.test.js (hunk 2 above)
Comment (dinodsaurus) [reply src/KiwiLogin/__test__/KiwiLogin.test.js]: yep, makes sense here 👍

File: src/KiwiLogin/__test__/KiwiLogin.test.js (hunk 3 above)
Comment (git-toni) [reply src/KiwiLogin/__test__/KiwiLogin.test.js]: Ofc, please ignore my question. It should be totally equivalent :sweat_smile:

**Pull request merged**